File: remote-config/server/src/main/java/br/com/leonardoferreira/poc/Application.java
package br.com.leonardoferreira.poc;

import java.util.Properties;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

}

@RestController
class MyController {

    @GetMapping("properties")
    Properties pingPong() {
        final Properties properties = new Properties();
        properties.put("custom.simple", "my-remote-value");
        return properties;
    }

}

File: feign/openfeign-micrometer/src/main/kotlin/br/com/leonardoferreira/poc/Application.kt
package br.com.leonardoferreira.poc

import feign.Feign
import feign.micrometer.MicrometerCapability
import io.micrometer.core.instrument.MeterRegistry
import io.micrometer.core.instrument.logging.LoggingMeterRegistry
import org.springframework.boot.ApplicationArguments
import org.springframework.boot.ApplicationRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.openfeign.EnableFeignClients
import org.springframework.cloud.openfeign.FeignBuilderCustomizer
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import java.time.Duration

@EnableFeignClients
@SpringBootApplication
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}

@Configuration
class MeterRegistryConfig {

    @Bean
    fun meterRegistry(): MeterRegistry =
        LoggingMeterRegistry()

}

@FeignClient(name = "githubClient", url = "https://api.github.com")
interface GithubClient {

    @GetMapping("/repos/{owner}/{repo}/contributors")
    fun findContributors(@PathVariable owner: String, @PathVariable repo: String): List<Contributor>

}

data class Contributor(val login: String, val contributions: Int)

@Configuration
class MicrometerFeignBuilderCustomizer(
    private val meterRegistry: MeterRegistry
) : FeignBuilderCustomizer {

    override fun customize(builder: Feign.Builder) {
        builder.addCapability(MicrometerCapability(meterRegistry))
    }

}

@Component
class Runner(
    private val githubClient: GithubClient
) : ApplicationRunner {

    override fun run(args: ApplicationArguments) {
        githubClient.findContributors("OpenFeign", "feign")
    }

}
